cleanup_test_artifacts.sh

```
@@ -0,0 +1,85 @@
+
+# 3. Clean htmlcov directory
+echo -e "\n${YELLOW}Cleaning coverage reports...${NC}"
+safe_remove "htmlcov"
+
+# 4. Clean test_data directory (be careful here)
+echo -e "\n${YELLOW}Cleaning test_data directory...${NC}"
+if [ -d "test_data" ]; then
+    # Count before cleaning
+    count=$(find test_data -type d -name "test_env_*" | wc -l)
+
+    if [ $count -gt 0 ]; then
```

Shellcheck (0.10.0)

[info] Double quote to prevent globbing and word splitting.

(SC2086)